File: crates/kernava-indexer/tests/fixtures/rb-small/calc.rb
require 'json'
require_relative 'helper'

class Calculator
  def initialize(value)
    @value = value
  end

  def add(x, y)
    compute(x)
  end

  private

  def helper(x)
    x * 2
  end

  def compute(x)
    helper(x) + @value
  end
end

module Math
  def compute(x)
    x * 2
  end
end

def free_function(a, b)
  a + b
end

File: crates/kernava-indexer/tests/fixtures/php-small/calc.php
<?php
namespace Example;

use Example\Math;

class Calculator {
    private $value;

    public function __construct($v) {
        $this->value = $v;
    }

    public function add($x, $y) {
        return $this->compute($x);
    }

    private function helper($x) {
        return $x * 2;
    }

    public function compute($x) {
        return $this->helper($x) + $this->value;
    }
}

interface Math {
    public function compute($x);
}

function free_function($a, $b) {
    return $a + $b;
}
